show/main.py

```diff
@@ -888,23 +892,213 @@ def get_bgp_peer():
+    # This prints out the show ip route header based on FRR 7.2 version.
+    # Please note that if we moved to future versions, we may heva to make changes to this
+    print("Codes: K - kernel route, C - connected, S - static, R - RIP,")
+    print("       O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,")
+    print("       T - Table, v - VNC, V - VNC-Direct, A - Babel, D - SHARP,")
+    print("       F - PBR, f - OpenFabric,")
+    print("       > - selected route, * - FIB route, q - queued route, r - rejected route\n")
+
 #
 # 'route' subcommand ("show ip route")
 #
 
 @ip.command()
 @click.argument('args', metavar='[IPADDRESS] [vrf <vrf_name>] [...]', nargs=-1, required=False)
+@multi_asic_util.multi_asic_click_options
 @click.option('--verbose', is_flag=True, help="Enable verbose output")
-def route(args, verbose):
+def route(args, namespace, display, verbose):
     """Show IP (IPv4) routing table"""
-    cmd = 'sudo vtysh -c "show ip route'
+    is_multi_asic = multi_asic.is_multi_asic()
+    if is_multi_asic:
+        # handling multi-ASIC by gathering the output from specified/all name space into a dictionary via 
+        # jason option and then filter out the json entries (by removing those next Hop that are 
+        # back-end ASIC if display is for front-end only). If the entry itself has no more next Hop after filtering, 
+        # then skip over that particular route entry.  Once completed, if the user chose "json" option, 
+        # then just print out the dictionary in Json format accordingly. But if no "json" option specified,
+        # then print out the header and the decoded entry representation for each route accordingly.
+        # This code is based on FRR 7.2 branch. If we moved to a new version we may need to change here as well
+        asic_cnt = multi_asic.get_num_asics()
+        arg_strg = ""
+        found_json = 0
+        back_end_intf_str = "PortChannel4"
+        if display == "all":
+            filter_back_end = False
+        else:
+            filter_back_end = True
+        if str(namespace) == "None":
+            start_range = 0
+            end_range = asic_cnt
+        else:
+            # -n asic0|asic1|...|asicN    So skip the first 4 chars to get the namespace number only
+            start_range = int(namespace[4:])
+            end_range = start_range + 1
+        hdr_printed = False
+        del_cnt = 0
+        # get all the other arguments except json that needs to be the last argument of the cmd if present
+        for arg in args:
+            arg_strg += " " + str(arg)
+            if str(arg) == "json":
+                found_json = 1
+        if not found_json:
+            arg_strg += " json"
+        for ns in range(start_range, end_range):
+            filtered_route = {}
+            # Need to add "ns" to form bgpX so it is sent to the correct bgpX docker to handle the request
+            cmd = 'sudo docker exec bgp' + str(ns) + ' vtysh -c "show ip route' + arg_strg + '"'
```

Comment:
please use .format() to create the string

Reply:
Agreed.  Have changed this to use .format to make it more readable